### PR TITLE
VUMIGO-291 create less vumi user apis.

### PR DESCRIPTION
**Pull request by smn**
We're creating too many of these for each request, just request the one
that's attached to the request object.

### CODE REVIEW

File: go/contacts/tests.py [hunk 1]
```diff
@@ -391,19 +393,19 @@ def test_contact_for_addr(self):
...
+            self.contact_store.contact_for_addr('twitter', u'@someone').key,
             twitter_contact.key)
         self.assertEqual(
-            self.contact_store.contact_for_addr('gtalk', 'gtalk@host.com').key,
-            gtalk_contact.key)
+            self.contact_store.contact_for_addr('gtalk',
+                u'gtalk@host.com').key, gtalk_contact.key)
         self.assertRaisesRegexp(RuntimeError, 'Unsupported transport_type',
-            self.contact_store.contact_for_addr, 'unknown', 'unknown')
+            self.contact_store.contact_for_addr, u'unknown', u'unknown')
```
Comment (hodgestar) [inline go/contacts/tests.py:408]: What are the test changes about? Were they failing before?

**Comment by jerith** [comment]
:+1:

**Comment by hodgestar** [comment]
:+1: modulo question about the test changes.

File: go/contacts/tests.py (hunk 1 above)
Comment (smn) [inline go/contacts/tests.py:408]: These were some errors that an earlier test with a protocol buffers transport raised. Sorry, not relevant to this commit & happy to remove them again if it's an issue.

Comment (hodgestar) [inline go/contacts/tests.py:408]: Happy for them to stay.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**